packages/web3/test/integration/main_web3.test.ts

```diff
@@ -84,6 +84,16 @@ describe('Web3 instance', () => {
 		}
 	});
 
+	it('should be able to create web3 object without provider', () => {
```

Comment:
I think we should also add another test with no provider web3 instantiation and using some functionality that is not dependent on provider.